src/main/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetails.java

```diff
@@ -741,8 +743,8 @@ public String validateGiverRecipientVisibility(FeedbackQuestionAttributes feedba
             for (String key : weightedPercentagePerOption.keySet()) {
                 int frequency = answerFrequency.get(key);
                 double weight = weightedPercentagePerOption.get(key);
-                double weightedPercentage =
-                        100 * ((frequency * weight) / calculateTotalWeightedResponseCount(answerFrequency));
+                double weightedPercentage = weight == 0 ? 0
```

Comment:
The check is redundant I think.

zero times a rational number is zero. zero divides rational number is also zero.

Reply:
The bug you mentioned in https://github.com/TEAMMATES/teammates/pull/8860#pullrequestreview-131410993 originates from here. When `totalWeightedResponseCount == 0`, this bug occurs. Though I can modify this check to test for  `totalWeightedResponseCount == 0` only. 

>zero divides rational number is also zero.

Umm. Isn't 0/0 `undefined`? That is why the bug is occuring. Though it is often confusing, I found this article which explains this nicely.
https://brilliant.org/wiki/what-is-0-0/

Reply:
Then your check should be done on `totalWeightedResponseCount` rather than `weight`

Reply:
Yes, that is what I am doing, I have changed the `weight` to `totalWeightedResponseCount`.